notebooks/contrib-dev/Filtration.ipynb

```diff
@@ -1,1030 +1,1316 @@
+        "    assert max_iter >= 0, \"Warning: 'max_iter' should be non-negative\"\n",
+        "\n",
+        "    x = x0\n",
+        "    if (LOUD):\n",
+        "        print(\"x0 =\",x0)\n",
+        "    iterations = 0\n",
+        "    converged = False\n",
+        "\n",
+        "    # Check if the residual is close enough to zero\n",
+        "    while (not converged and iterations < max_iter):\n",
+        "\n",
+        "        if (LOUD):\n",
+        "            print(\"x_\",iterations+1,\"=\",x,\"-\",f(x),\"/\",\n",
+        "                  fprime(x),\"=\",x - f(x)/fprime(x))\n",
+        "    ### BEGIN SOLUTION\n",
+        "        #single-variable Newton's method equation\n",
+        "        x = x - f(x)/fprime(x)\n",
+        "    ### END SOLUTION\n",
+        "        # check if converged\n",
+        "        if np.fabs(f(x)) < epsilon:\n",
+        "            converged = True\n",
+        "\n",
+        "        iterations += 1\n",
+        "    print(\"It took\",iterations,\"iterations\")\n",
+        "\n",
+        "    if not converged:\n",
+        "        print(\"Warning: Not a solution. Maximum number of iterations exceeded.\")\n",
+        "    return x #return estimate of root"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "asuUuhfuvNh8"
+      },
+      "source": [
+        "Using these functions, determine what the Ruth constant in SI units `K_SI` will be when the pressure drop $ΔP$ is $0.002 N/m^{2}$.\n",
+        "> **Hint**: Begin by making two functions to satisfy the terms of the Newton's method function: `Nonlinear_function`, which will return the canonical form of the equation we used above to solve for $ΔP$, and `Dnonlinear_function`, which returns the derivative of `Nonlinear function` with respect to the variable being solved for (`K_SI`)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "qJaTNCvLvhcn",
+        "outputId": "29e6a751-6412-402b-cf1a-da1ba5a6f0ff"
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "x0 = 1\n",
+            "x_ 1 = 1 - -14.491960849645977 / -14.493960849645978 = 0.00013798850574719967\n",
+            "It took 1 iterations\n",
+            "0.00013798850574719967\n"
+          ]
+        }
+      ],
+      "source": [
+        "deltaP = 0.002 #N/m^2\n",
+        "\n",
+        "#Constants for system\n",
+        "alpha = 4 #m/kg\n",
+        "C = 1920 #kg/m^3\n",
+        "mu = 2.9E-3 #kg/m*s\n",
+        "A = 0.28 #m^2\n",
+        "gc = 9.8 #kgf / kg*m *s^2\n",
+        "\n",
+        "### BEGIN SOLUTION\n",
+        "#Define pressure drop formula in canonical form\n",
+        "def Dnonlinear_function(x):\n",
```

Comment:
Do not capitalize function names in Python by convention. Change this to d_nonlinear_function and nonlinear_function